File: newton.py
import math

def newton(f, df, xo, n):
    for _ in range(n):
        xo = xo - f(xo) / df(xo)
        yield xo

def f(x):
    return x * (x - 1) * (x - 2) + 0.42

def df(x):
    return 3 * x ** 2 - 6 * x + 2

xo = 2.31658864
n = 200

results = list(newton(f, df, xo, n))

iterations = [5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 100, 105, 110, 115, 120, 125, 130, 135, 140, 145, 150, 155, 160, 165, 170, 175, 180, 185, 190, 195]

for i in iterations:
    print(results[i - 1], ",")

File: newton.c
#include <stdio.h>
#include <math.h>

void newton(double (*f)(double), double (*df)(double), double xo, int n)
{
    int iterations[] = {1, 25, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700};
    int j = 0;
    
    for (int i = 0; i < n; i++)
    {

        if (df(xo) == 0)
        {
            break;
        }

        double xi = xo - f(xo) / df(xo);
        // printf("x_%i = %.16f\n", i + 1, xi);

        if (i + 1 == iterations[j])
        {
            printf("%.16f,\n", xi);
            j++;
        }

        xo = xi;
    }
}

void main()
{
    double f(double x)
    {
        return exp(5 * x) - 2;
    }

    double df(double x)
    {
        return 5 * exp(5 * x);
    }

    double xo = -1.16391525;
    int n = 800;

    newton(f, df, xo, n);
}